### DISCUSSION TITLE
Does Maddy Build for Windows?

### DISCUSSION THREAD

**Discussion by lonnietc** [emersion/maddy#481] [Q&A] [answered]
Hello,

Just searching for a full Golang email server that can also build for Windows, Linux, etc.

Can Maddy?
Thanks

**Comment by foxcpp** [comment, marked as answer]
Yes, maddy can be compiled for Windows. It will run without any problems on any Windows version that supports Unix sockets (Windows 10) and with some limitations on any other Windows system.

The only current issue is missing proper default paths for state_dir and runtime_dir (#290) so you would need to set these appropriately in the configuration file. Also, build.sh is obviously not useful, you would need to compile the executable directly:
```
go build ./cmd/maddy
go build ./cmd/maddyctl
```
For it to run correctly, you need Go and MinGW toolchains installed and configured appropriately.

**Comment by lonnietc** [comment]
Thanks for getting back to me and I will work on compiling it up.

What seems to be interesting is that I think that Maddy also has IMAP and POP3 support as well, if I read things correctly.

What I am interested in doing is to build it into a multi-platform P2P framework that also supports calling standard email servers. This would allow users to have a decentralized P2P email network without the need for centralized email servers which could be run simply and easily on their desktop systems and connecting to it via just about any clients (Thunderbird, etc.) that they choose.

That's the idea, at least.
Thanks again
